### ISSUE TITLE
[Weather] There is too much rain

### ISSUE THREAD

**Issue by Rasgnarok** [SpongePowered/Sponge#1860]
**I am currently running**

- SpongeForge version: 2990
- SpongeVanilla version: N/A
- Java version: 1.8
- Operating System: Centos 7

- Plugins/Mods: 
![image](https://user-images.githubusercontent.com/18584380/38448429-163dceec-39d2-11e8-9a7c-53d722fc9dac.png)

**Issue Description**
- Locking weather won't stop the rain from coming back (through `/nucleus:lockweather`)
- Only way to stop the rain from coming back is turning `doWeathecycle` false

**Comment by Inscrutable** [comment]
I've been having the same issue on every build of SpongeForge for 1.12.2 that I've tried. The nucleus commands to set the weather to clear work, but will not persist. I set to clear for 40 hours, yet it rained again in two minutes. Toggledownfall seems to revert very quickly as well. It likes to rain.

**Comment by Lemmmy** [comment]
duplicate of SpongePowered/SpongeForge#1138? several server admins reporting too much rain there

**Comment by bloodmc** [comment]
@Meronat Please look into this as you were the last one to change related code.

https://github.com/SpongePowered/SpongeCommon/commit/534ccd511878f2f8a38b8ce9b321b656154d057b

**Comment by parlough** [comment]
Will look into it.

**Comment by parlough** [comment]
Faith and I have a fix which I think we'll fix both issues mentioned here as well as other issues around weather which may be present. Will test a bit more then push it.

**Comment by parlough** [comment]
So https://github.com/SpongePowered/SpongeCommon/pull/1866 should fix the issues you're having with the lockweather command as well as some other event issues. 

I tested extensively with these changes and I didn't see it raining any longer than it should. Rain can last anywhere for half through a whole Minecraft day(10-20 minutes) and it can start raining again as often as every .5-7.5 Minecraft days(10-10 minutes) when letting it run naturally. This can be quite often and it may be larger than it used to be a long time ago, so maybe it just feels like quite a lot. 

Nucleus's weather command was a bit wrong as well I think and thought it was setting weather for x seconds when it was really lasting for that many ticks instead, so this could be another source of the shorter clear times you were experiencing. 

We were also sending the wrong packets once when the weather change event was cancelled, but I don't think that should have been a source of your issues, but it may have somehow, so I have fixed that as well.
